File: examples/openapi-sdk-petstore/kotlin/sdk/com/example/petstore/api/PetApi.kt
package com.example.petstore.api

import com.example.petstore.models.*
import okhttp3.Response

public interface PetApi {
    /** POST /pet */
    suspend fun addPet(
        body: Pet,
        options: RequestOptions
    ): Pet

    /** POST /pet */
    suspend fun addPetWithResponse(
        body: Pet,
        options: RequestOptions
    ): Pair<Pet, Response>

    /** PUT /pet */
    suspend fun updatePet(
        body: Pet,
        options: RequestOptions
    ): Pet

    /** PUT /pet */
    suspend fun updatePetWithResponse(
        body: Pet,
        options: RequestOptions
    ): Pair<Pet, Response>

    /** GET /pet/findByStatus */
    suspend fun findPetsByStatus(
        status: FindPetsByStatus_Param_Status,
        options: RequestOptions
    ): List<Pet>

    /** GET /pet/findByStatus */
    suspend fun findPetsByStatusWithResponse(
        status: FindPetsByStatus_Param_Status,
        options: RequestOptions
    ): Pair<List<Pet>, Response>

    /** GET /pet/findByTags */
    suspend fun findPetsByTags(
        tags: List<String>,
        options: RequestOptions
    ): List<Pet>

    /** GET /pet/findByTags */
    suspend fun findPetsByTagsWithResponse(
        tags: List<String>,
        options: RequestOptions
    ): Pair<List<Pet>, Response>

    /** GET /pet/{petId} */
    suspend fun getPetById(
        petId: Long,
        options: RequestOptions
    ): Pet

    /** GET /pet/{petId} */
    suspend fun getPetByIdWithResponse(
        petId: Long,
        options: RequestOptions
    ): Pair<Pet, Response>

    /** POST /pet/{petId} */
    suspend fun updatePetWithForm(
        petId: Long,
        name: String?,
        status: String?,
        options: RequestOptions
    ): Pet

    /** POST /pet/{petId} */
    suspend fun updatePetWithFormWithResponse(
        petId: Long,
        name: String?,
        status: String?,
        options: RequestOptions
    ): Pair<Pet, Response>

    /** DELETE /pet/{petId} */
    suspend fun deletePet(
        petId: Long,
        apiKey: String?,
        options: RequestOptions
    )

    /** DELETE /pet/{petId} */
    suspend fun deletePetWithResponse(
        petId: Long,
        apiKey: String?,
        options: RequestOptions
    ): Response

    /** POST /pet/{petId}/uploadImage */
    suspend fun uploadFile(
        petId: Long,
        additionalMetadata: String?,
        body: ByteArray,
        options: RequestOptions
    ): ApiResponse

    /** POST /pet/{petId}/uploadImage */
    suspend fun uploadFileWithResponse(
        petId: Long,
        additionalMetadata: String?,
        body: ByteArray,
        options: RequestOptions
    ): Pair<ApiResponse, Response>

    /** POST /pet/{petId}/uploadDocument */
    suspend fun uploadPetDocument(
        petId: Long,
        file: ByteArray,
        title: String?,
        description: String?,
        options: RequestOptions
    ): ApiResponse

    /** POST /pet/{petId}/uploadDocument */
    suspend fun uploadPetDocumentWithResponse(
        petId: Long,
        file: ByteArray,
        title: String?,
        description: String?,
        options: RequestOptions
    ): Pair<ApiResponse, Response>

    /** POST /tags */
    suspend fun submitTags(
        body: SubmitTags_Body,
        options: RequestOptions
    ): SubmitTags_Response

    /** POST /tags */
    suspend fun submitTagsWithResponse(
        body: SubmitTags_Body,
        options: RequestOptions
    ): Pair<SubmitTags_Response, Response>
}

File: examples/openapi-sdk-petstore/swift/sdk/API/PetAPI.swift
import Foundation

public protocol PetAPI {
    /// POST /pet
    func addPet(
        body: Pet,
        options: RequestOptions
    ) async throws -> Pet

    /// POST /pet
    func addPetWithResponse(
        body: Pet,
        options: RequestOptions
    ) async throws -> (Pet, HTTPURLResponse)

    /// PUT /pet
    func updatePet(
        body: Pet,
        options: RequestOptions
    ) async throws -> Pet

    /// PUT /pet
    func updatePetWithResponse(
        body: Pet,
        options: RequestOptions
    ) async throws -> (Pet, HTTPURLResponse)

    /// GET /pet/findByStatus
    func findPetsByStatus(
        status: FindPetsByStatus_Param_Status,
        options: RequestOptions
    ) async throws -> [Pet]

    /// GET /pet/findByStatus
    func findPetsByStatusWithResponse(
        status: FindPetsByStatus_Param_Status,
        options: RequestOptions
    ) async throws -> ([Pet], HTTPURLResponse)

    /// GET /pet/findByTags
    func findPetsByTags(
        tags: [String],
        options: RequestOptions
    ) async throws -> [Pet]

    /// GET /pet/findByTags
    func findPetsByTagsWithResponse(
        tags: [String],
        options: RequestOptions
    ) async throws -> ([Pet], HTTPURLResponse)

    /// GET /pet/{petId}
    func getPetById(
        petId: Int64,
        options: RequestOptions
    ) async throws -> Pet

    /// GET /pet/{petId}
    func getPetByIdWithResponse(
        petId: Int64,
        options: RequestOptions
    ) async throws -> (Pet, HTTPURLResponse)

    /// POST /pet/{petId}
    func updatePetWithForm(
        petId: Int64,
        name: String?,
        status: String?,
        options: RequestOptions
    ) async throws -> Pet

    /// POST /pet/{petId}
    func updatePetWithFormWithResponse(
        petId: Int64,
        name: String?,
        status: String?,
        options: RequestOptions
    ) async throws -> (Pet, HTTPURLResponse)

    /// DELETE /pet/{petId}
    func deletePet(
        petId: Int64,
        apiKey: String?,
        options: RequestOptions
    ) async throws

    /// DELETE /pet/{petId}
    func deletePetWithResponse(
        petId: Int64,
        apiKey: String?,
        options: RequestOptions
    ) async throws -> HTTPURLResponse

    /// POST /pet/{petId}/uploadImage
    func uploadFile(
        petId: Int64,
        additionalMetadata: String?,
        body: Data,
        options: RequestOptions
    ) async throws -> ApiResponse

    /// POST /pet/{petId}/uploadImage
    func uploadFileWithResponse(
        petId: Int64,
        additionalMetadata: String?,
        body: Data,
        options: RequestOptions
    ) async throws -> (ApiResponse, HTTPURLResponse)

    /// POST /pet/{petId}/uploadDocument
    func uploadPetDocument(
        petId: Int64,
        file: Data,
        title: String?,
        description: String?,
        options: RequestOptions
    ) async throws -> ApiResponse

    /// POST /pet/{petId}/uploadDocument
    func uploadPetDocumentWithResponse(
        petId: Int64,
        file: Data,
        title: String?,
        description: String?,
        options: RequestOptions
    ) async throws -> (ApiResponse, HTTPURLResponse)

    /// POST /tags
    func submitTags(
        body: SubmitTags_Body,
        options: RequestOptions
    ) async throws -> SubmitTags_Response

    /// POST /tags
    func submitTagsWithResponse(
        body: SubmitTags_Body,
        options: RequestOptions
    ) async throws -> (SubmitTags_Response, HTTPURLResponse)
}
